### PR TITLE
prototype demonstrating d3fc-scripts (create-d3-module)

### PR DESCRIPTION
**Pull request by ColinEberhardt**
This pull request is a quick prototype to show the use of d3fc-scripts, a create-react-app style approach to d3 modules, as discussed here: https://github.com/d3/d3/issues/3116

The general concept is to centralise common infrastructure (build, test, lint, ...) into a single package for easier maintenance and the creation of new modules without cut-and-paste.

We've been using this approach within [d3fc](https://github.com/d3fc) for a little while, via [d3fc-scripts](https://github.com/d3fc/d3fc-scripts). While some of our choices differ from those in d3 (jasmine vs tape, ES201* vs ES2015) the general principles are the same.

I've created a branch of d3fc-scripts, tailoring for the current d3 projects, as a proof of concept. This PR demonstrates the following:

 - The removal of common infrastructure dependencies from d3-axis
 - The removal of common build scripts from d3-axis

d3fc-scripts performs the following:
 - bundling, using rollup and uglify
 - linting
 - unit test execution
 - static checks for project meta-data (licence files, npmignore etc ...)

Basically anything that can be centralised is moved to d3fc-scripts, anything that cannot is checked via the tests.

If this looks like a good approach, I'd propose to:

 - resolve the differences between d3fc and d3 in terms of testing and JavaScript syntax, either by making d3fc-scripts configurable, or updating the d3fc modules to match d3
 - look at the publishing step, for d3fc we use semantic-release
 - document d3fc-scripts!
 - make sure it works for other module authors.

### CODE REVIEW

**Comment by mbostock** [comment]
Thanks for the contribution and apologies for the delay. I’m not planning on making any major changes to the D3 modules at the moment.
